File: umbrella-insurance-react-ui/src/models/people/v1/Person.ts
export class Person {
    id?: number;
    firstName?: string;
    middleName?: string;
    surname?: string;
    dateOfBirth?: string; //yyyy-mm-dd
    ssn?: string;

    constructor(obj?: any) {
        if (obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if (obj?.firstName) {
                this.firstName = obj.firstName;
            }
            if (obj?.middleName) {
                this.middleName = obj.middleName;
            }
            if (obj?.surname) {
                this.surname = obj.surname;
            }
            if (obj?.dateOfBirth) {
                this.dateOfBirth = obj.dateOfBirth;
            }
            if (obj?.ssn) {
                this.ssn = obj.ssn;
            }
        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Person.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.DateDeserializer;
import com.umbrella.insurance.core.serializers.DateSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.sql.Date;

@ToString
@Getter
@Setter
@Entity
@Table(name = "people", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "people_ssn_unique", columnNames = {"ssn"})
})
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "person_id", nullable = false)
    private Long id;

    @NotNull
    @Column(name = "first_name", nullable = false, length = Integer.MAX_VALUE)
    private String firstName;

    @Column(name = "middle_name", length = Integer.MAX_VALUE)
    private String middleName;

    @NotNull
    @Column(name = "surname", nullable = false, length = Integer.MAX_VALUE)
    private String surname;

    @JsonSerialize(using = DateSerializer.class)
    @JsonDeserialize(using = DateDeserializer.class)
    @NotNull
    @Column(name = "date_of_birth", nullable = false)
    private Date dateOfBirth;

    @Column(name = "ssn", length = Integer.MAX_VALUE)
    private String ssn;

}